accounts/abi/bind/bind.go

```diff
@@ -111,11 +120,20 @@ func Bind(types []string, abis []string, bytecodes []string, pkg string, lang La
 					if input.Name == "" {
 						normalized.Inputs[j].Name = fmt.Sprintf("arg%d", j)
 					}
+					if _, exist := structs[input.Type.String()]; input.Type.T == abi.TupleTy && !exist {
+						bindStructType[lang](input.Type, structs)
+					}
 				}
 			}
 			// Append the event to the accumulator list
 			events[original.Name] = &tmplEvent{Original: original, Normalized: normalized}
 		}
+
+		// There is no easy way to pass arbitrary java objects to the Go side.
```

Comment:
you could destructure the object using the whole `{{range}}` but let's not stall this PR any longer and work on it with ligi when he gets back.

Reply:
Yes. Actually, I already have a few hacks in my local for Java binding. Now I can support structs in Java, but not struct slice(can't really remember). But it's really hacky so I don't push it.